core/src/main/java/org/elasticsearch/index/query/FilteredQueryBuilder.java

```diff
@@ -31,36 +36,113 @@
 @Deprecated
 public class FilteredQueryBuilder extends AbstractQueryBuilder<FilteredQueryBuilder> {
 
+    /** Name of the query in the REST API. */
     public static final String NAME = "filtered";
-
+    /** The query to filter. */
     private final QueryBuilder queryBuilder;
-
+    /** The filter to apply to the query. */
     private final QueryBuilder filterBuilder;
 
     static final FilteredQueryBuilder PROTOTYPE = new FilteredQueryBuilder(null, null);
 
     /**
+     * Returns a {@link MatchAllQueryBuilder} instance that will be used as
+     * default queryBuilder if none is supplied by the user. Feel free to
+     * set queryName and boost on that that instance - it's always a new one.
+     * */
+    public static QueryBuilder generateDefaultQuery() {
```

Comment:
a constant could cause problems though if the constant gets misused (e.g. changing its boost etc.), method is ok sorry